### PR TITLE
2582 fwreznetz 613 add csv import interface for app server ip addresses in UI

### LINKED ISSUE #2582: FWREZNETZ-613 - Add CSV-Import interface for app server IP addresses in UI
When modelling as admin role user, the menu "app server" should be enhanced so that the admin can also upload and import CSV files of the following format;

- External App-ID
- App Server Type
- App-IP-Address-Range (Start-IP+ optional End-IP) 

- [x] Add csv format to help file/icon tooltip
- [x] Display subnet  or ip range at App Server table in app server ui
- [x] Display UI message on successfull import

Duplicate IP addresses (that already exist for the application) should be ignored.

### PR DESCRIPTION
**Pull request by SolidProgramming**
Closes #2582

### CODE REVIEW

**Comment by tpurschke** [comment]
@SolidProgramming can you have a look at the conflicts, please?

File: roles/database/files/sql/idempotent/fworch-texts.sql [hunk 1]
```diff
@@ -2425,7 +2427,8 @@ INSERT INTO txt VALUES ('general',              'German', 	'Allgemein');
 INSERT INTO txt VALUES ('general',              'English', 	'General');
 INSERT INTO txt VALUES ('naming_convention',    'German',   'Namenskonvention');
 INSERT INTO txt VALUES ('naming_convention',    'English',  'Naming Convention');
-
+INSERT INTO txt VALUES ('import_app_server',    'German',   'App Server importieren');
+INSERT INTO txt VALUES ('import_app_server',    'English',  'Import App Server');
```
Comment (tpurschke) [inline roles/database/files/sql/idempotent/fworch-texts.sql:2431]: should be: Import app servers (plural)

**Review by tpurschke — CHANGES_REQUESTED** [review]
ALTER TABLE owner_network ADD CONSTRAINT owner_network_ip_unique UNIQUE (owner_id, ip);

should be

    ALTER TABLE owner_network ADD CONSTRAINT owner_network_ip_unique UNIQUE (owner_id, ip, ip_end);

and (needs to be tested) in roles/lib/files/FWO.Api.Client/APIcalls/modelling/upsertAppServer.graphql:

      on_conflict: {
          constraint: owner_network_ip_unique

File: roles/ui/files/FWO.UI/Data/CSVAppServerImportModel.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace FWO.Ui.Data
+{
+    public class CSVAppServerImportModel
+    {
+        public string? AppServerName { get; set; }
+        public string? AppID { get; set; }
+        public string? AppServerTyp { get; set; }
+        public string? AppIPRange { get; set; }
```
Comment (tpurschke) [inline roles/ui/files/FWO.UI/Data/CSVAppServerImportModel.cs:8]: AppIPRangeStart instead?

File: roles/ui/files/FWO.UI/Services/ModellingAppServerHandler.cs [hunk 3]
```diff
@@ -87,12 +87,25 @@ private async Task<bool> AddAppServerToDb()
...
             {
+                if (ActAppServer.Ip.TrySplit('-', 1, out string ipEnd))
+                {
+                    ActAppServer.Ip = ActAppServer.Ip.Split('-')[0];
+                    ActAppServer.IpEnd = ipEnd;
+                }
+
+                if (!ActAppServer.Ip.TrySplit('/', 1, out string? subnet))
+                {
+                    ActAppServer.Ip += "/32";
```
Comment (tpurschke) [inline roles/ui/files/FWO.UI/Services/ModellingAppServerHandler.cs:98]: this needs to be adjusted to not only work for ipv4 (/32) but also for ipv6 (/128)

File: roles/ui/files/FWO.UI/Shared/FileUpload.razor [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+            if (appServer.AppIPRange.TrySplit('-', 1, out string ipEnd))
+            {
+                appServer.AppIPRange = appServer.AppIPRange.Split('-')[0];
+                appServer.AppIPRangeEnd = ipEnd;
+            }
+
+            if (string.IsNullOrEmpty(appServer.AppIPRangeEnd) && !appServer.AppIPRange.TrySplit('/', 1, out string subnet))
+            {
+                appServer.AppIPRange += "/32";
```
Comment (tpurschke) [inline roles/ui/files/FWO.UI/Shared/FileUpload.razor:127]: this needs to be adjusted to not only work for ipv4 (/32) but also for ipv6 (/128)

File: roles/ui/files/FWO.UI/Services/ModellingAppServerHandler.cs (hunk 3 above)
Comment (SolidProgramming) [reply roles/ui/files/FWO.UI/Services/ModellingAppServerHandler.cs:98]: oh right...

File: roles/ui/files/FWO.UI/Shared/FileUpload.razor (hunk 4 above)
Comment (SolidProgramming) [reply roles/ui/files/FWO.UI/Shared/FileUpload.razor:127]: oh right...

**Review by tpurschke — CHANGES_REQUESTED** [review]
just tested importing two subnet data files and unfortunately "the second one wins".
should actually be additive - importing all files and after that deciding which NAs and subnets have been deleted.

**Comment by tpurschke** [comment]
- I still got the error when first visiting settings modelling (goes away when first saving)
- but all my settings regarding naming conventions where gone after saving

**Comment by SolidProgramming** [comment]
@tpurschke 
```Csharp
  public async Task TestSelectAppServersFromArea()
  {
      List<ModellingAppServer> expectedResult =
      [
          new(AppServerInside1) { TooltipText = userConfig.GetText("C9002") },
          new(AppServerInside2) { TooltipText = userConfig.GetText("C9002") },
          new(AppServerInside3) { TooltipText = userConfig.GetText("C9002") }
      ];
      await AppRoleHandler.SelectAppServersFromArea(TestArea);
      ClassicAssert.AreEqual(expectedResult, AppRoleHandler.AppServersInArea);
  }
```

yeag it fails on my install too

**Review by tpurschke — APPROVED** [review]
still some issues but merging for now

**Pull request merged**